### PR TITLE
feat: restructure navigation for console

### PR DESCRIPTION
**Pull request by wesbillman**
- Restructure console navigation to better support future updates
- Add new "modules" and "infrastructure" sections
- Add sample tree to "modules"

Screenshot 2024-08-23 at 11 43 20 AM
Screenshot 2024-08-23 at 11 43 22 AM
Screenshot 2024-08-23 at 11 43 28 AM
Screenshot 2024-08-23 at 11 43 31 AM
Screenshot 2024-08-23 at 11 43 33 AM
Screenshot 2024-08-23 at 11 43 37 AM
Screenshot 2024-08-23 at 11 43 41 AM

### CODE REVIEW

File: frontend/src/features/modules/module.utils.ts [hunk 1]
```diff
@@ -1,4 +1,16 @@
...
+  ArrowDownOnSquareStackIcon,
+  BellAlertIcon,
+  BoltIcon,
+  CircleStackIcon,
+  CodeBracketIcon,
+  CogIcon,
+  LockClosedIcon,
+  RectangleGroupIcon,
+} from '@heroicons/react/24/outline'
+import type { ForwardRefExoticComponent, SVGProps } from 'react'
```
Comment (deniseli) [inline frontend/src/features/modules/module.utils.ts:11]: lol exotic component is such a good name

**Review by deniseli — APPROVED** [review]
gorgeous

File: frontend/src/features/modules/module.utils.ts (hunk 1 above)
Comment (wesbillman) [reply frontend/src/features/modules/module.utils.ts:11]: So exotic 😂

**Comment by alecthomas** [comment]
Noice

**Pull request merged**